File: lib/src/main/java/com/tomady/nutrition/data/local/diet/DishHistory.java
package com.tomady.nutrition.data.local.diet;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "dish_history")
public class DishHistory {
    @PrimaryKey(autoGenerate = true)
    private int id;
    private int userId;
    private int dishId;
    private String consumedAt;

    public DishHistory() {}

    public int getId() { return id; }
    public void setId(int id) { this.id = id; }
    public int getUserId() { return userId; }
    public void setUserId(int userId) { this.userId = userId; }
    public int getDishId() { return dishId; }
    public void setDishId(int dishId) { this.dishId = dishId; }
    public String getConsumedAt() { return consumedAt; }
    public void setConsumedAt(String consumedAt) { this.consumedAt = consumedAt; }
}

File: app/src/main/java/com/tomady/nutrition/data/local/diet/entity/DishHistory.kt
package com.tomady.nutrition.data.local.diet.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import androidx.room.PrimaryKey

/**
 * Room entity representing a logged meal event — what the user ate, when, and how much.
 */
@Entity(
    tableName = "dish_history",
    foreignKeys = [
        ForeignKey(
            entity = User::class,
            parentColumns = ["id"],
            childColumns = ["user_id"],
            onDelete = ForeignKey.CASCADE
        ),
        ForeignKey(
            entity = Dish::class,
            parentColumns = ["id"],
            childColumns = ["dish_id"],
            onDelete = ForeignKey.SET_NULL
        )
    ],
    indices = [Index("user_id"), Index("dish_id"), Index("date")]
)
data class DishHistory(
    @PrimaryKey
    @ColumnInfo(name = "id")
    val id: String,

    @ColumnInfo(name = "user_id")
    val userId: String,

    @ColumnInfo(name = "dish_id")
    val dishId: String? = null,

    @ColumnInfo(name = "date")
    val date: String,

    @ColumnInfo(name = "meal_type")
    val mealType: String? = null,

    @ColumnInfo(name = "servings")
    val servings: Double = 1.0,

    @ColumnInfo(name = "notes")
    val notes: String? = null,

    @ColumnInfo(name = "created_at")
    val createdAt: Long = System.currentTimeMillis(),

    @ColumnInfo(name = "updated_at")
    val updatedAt: Long = System.currentTimeMillis()
)
